packages/bug-detectors/internal/prototype-pollution.ts

```diff
@@ -0,0 +1,565 @@
+			// 	PrototypePollution.detectPrototypePollution(a, "a");
+			// 	return _jazzerPP_result0;
+			// })();"
+			// This expression will be further instrumented by the regular Jazzer.js instrumentation plugins.
+			AssignmentExpression(path: NodePath<types.AssignmentExpression>) {
+				if (!config) {
+					return;
+				}
+				if (
+					config &&
+					!config.getInstrumentAssignmentsAndVariableDeclarations()
+				) {
+					return;
+				}
+				if (
+					instrumentationGuard.has(
+						TrackedExpressions.ASSIGNMENT_EXPRESSION,
+						path.node
+					)
+				) {
+					return;
+				}
+				// Get identifier of the variable being assigned to
+				const identifier = getIdentifierFromAssignmentExpression(path.node);
+				if (!identifier) return;
+
+				// Wrap the whole expression in a lambda and check for __proto__ changes
+				const result = path.scope.generateUidIdentifier("jazzerPP_result");
+				instrumentationGuard.add(
+					TrackedExpressions.ASSIGNMENT_EXPRESSION,
+					path.node
+				);
+				// Copy path.node because it will be replaced by a lambda.
+				const resultDeclarator = types.variableDeclarator(
+					result,
+					JSON.parse(JSON.stringify(path.node))
+				);
+				instrumentationGuard.add(
+					TrackedExpressions.ASSIGNMENT_EXPRESSION,
+					resultDeclarator
+				);
+				instrumentationGuard.add(
+					TrackedExpressions.VARIABLE_DECLARATION,
+					resultDeclarator
+				);
+
+				const newAssignment = types.callExpression(
+					types.arrowFunctionExpression(
+						[],
+						types.blockStatement([
+							types.variableDeclaration("const", [resultDeclarator]),
+							// check for __proto__ changes
+							types.expressionStatement(
+								types.callExpression(
+									types.identifier(
+										"PrototypePollution.detectPrototypePollution"
+									),
+									[identifier, types.stringLiteral("" + identifier.name)]
+								)
+							),
+							// return the original assignment
+							types.returnStatement(result),
+						])
+					),
+					[]
+				);
+				path.replaceWith(newAssignment);
+			},
+			// Wraps variable declaration in a lambda, and checks if __proto__ of the identifier contains any non-function properties.
+			// For example: "const a = 10;" will be transpiled to:
+			// "const a = (() => {
+			//           const _jazzerPP0 = 10;
+			//           PrototypePollution.detectPrototypePollution(_jazzerPP0, "a");
+			//           return _jazzerPP0;
+			//         })();"
+			// This expression will be further instrumented by the regular Jazzer.js instrumentation plugins.
+			VariableDeclarator(path: NodePath<types.VariableDeclarator>) {
+				if (!config) {
+					return;
+				}
+				if (
+					config &&
+					!config.getInstrumentAssignmentsAndVariableDeclarations()
+				) {
+					return;
+				}
+				// wrap the initializer in a lambda and check for __proto__ changes
+				if (path.node.init) {
+					if (
+						instrumentationGuard.has(
+							TrackedExpressions.VARIABLE_DECLARATION,
+							path.node
```

Comment:
We call `has` on `path.node` here, but as far as I can tell never `add` it. Is another expression added below equal to it?

Would it make sense to replace `has` with something like `hasOrAdd`, which ensures that the value is always added if it isn't present?